templates/az-oci-exa-pdb/main.tf

```diff
@@ -3,33 +3,50 @@ resource "azurerm_resource_group" "resource_group" {
   name     = var.resource_group_name
 }
 
-module "vm_cluster_network" {
-  source = "../../modules/azure-vnet-subnet"
-  providers = {
-    azurerm = azurerm
+# Delete comment before merge
+# Reference code : https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/blob/5b60a9a29fa471d3298427f706645e80f09de9da/examples/legacy_address_prefix/README.md 
+module "avm_vmc_network" {
+  source = "Azure/avm-res-network-virtualnetwork/azurerm"
+  version = "0.2.4"
+
+  address_space       = var.virtual_network_address_space
+  location            = var.location
+  name                = var.virtual_network_name
+  resource_group_name = azurerm_resource_group.resource_group.name
+
+  subnets = {
+    delegated = {
+      name             = var.delegated_subnet_name
+      address_prefixes = var.delegated_subnet_address_prefix
+
+      delegation = [{
+        name = "Oracle.Database/networkAttachments"
+        service_delegation = {
+          name    = "Oracle.Database/networkAttachments"
+          actions = ["Microsoft.Network/networkinterfaces/*", "Microsoft.Network/virtualNetworks/subnets/join/action"]
+
+        }
+      }]
+    }
   }
-  location                        = var.location
-  resource_group_name             = azurerm_resource_group.resource_group.name
-  virtual_network_name            = var.virtual_network_name
-  delegated_subnet_address_prefix = var.delegated_subnet_address_prefix
-  virtual_network_address_space   = var.virtual_network_address_space
-  delegated_subnet_name           = var.delegated_subnet_name
 }
 
 module "exa_infra_and_vm_cluster" {
   source = "../../modules/azure-exainfra-vmcluster"
   providers = {
     azapi = azapi
   }
+  depends_on = [module.avm_vmc_network]
+
   exadata_infrastructure_resource_display_name                     = var.exadata_infrastructure_resource_display_name
   exadata_infrastructure_resource_name                             = var.exadata_infrastructure_resource_name
   location                                                         = var.location
-  oracle_database_delegated_subnet_id                              = module.vm_cluster_network.delegated_subnet_id
+  oracle_database_delegated_subnet_id                              = module.avm_vmc_network.subnets[0].id
```

Comment:
subnets is a map
`module.avm_vmc_network.subnets["delegated"].id`

Reply:
I tested `terraform plan` successfully with 
```
oracle_database_delegated_subnet_id = module.avm_vmc_network.subnets.delegated.resource_id
```

Reply:
I ran apply and it failed due to module.avm_vmc_network.subnets[0].id

Reply:
Can you please  pulled latest, after which you should see value as `module.avm_vmc_network.subnets.delegated.resource_id` not  `.subnets[0].id`